lib/vfscore/automount.c

```diff
@@ -78,11 +78,57 @@ struct vfscore_volume {
 };
 
 #if CONFIG_LIBVFSCORE_FSTAB
+/* Handle `mkpath` Unikraft Mount Option */
+static int vfscore_volume_process_ukopts_do_mkpath(const char *path)
+{
+	const char *curri;
+	char *pos;
+	int rc;
+
+	UK_ASSERT(path);
+
+	curri = path;
+	pos = strchr(path + 1, '/');
+
+	while (pos) {
+		/* If we reached last directory, delete trailing `/` and stop */
+		if (pos[1] == '\0') {
+			*pos = '\0';
+			break;
+		}
+
+		/* Zero out the next '/' */
+		*pos = '\0';
+
+		/* mkdir() with S_IRWXU */
+		rc = mkdir(curri, 0700);
+		if (rc && errno != EEXIST)
+			return rc;
+
+		/* Restore current '/' */
+		*pos = '/';
+
+		/* Now prepare for the next '/' in the path, if there */
+		pos = strchr(pos + 1, '/');
+	}
+
+	/* We have reached the last directory in the path */
+	rc = mkdir(curri, 0700);
+	if (rc && errno != EEXIST)
+		return rc;
+
+	return 0;
+}
+
 /**
  * vv->ukopts must follow the pattern below, each option separated by
  * the character defined through LIBVFSCORE_FSTAB_VOLUME_ARGS_SEP (e.g. with
  * LIBVFSCORE_FSTAB_VOLUME_ARGS_SEP = ':'):
  *	[<ukopt1>:<ukopt2>:<ukopt3>:...:<ukoptN>]
+ *
+ * Currently implemented, Unikraft Mount options:
+ * - mkpath	A custom mount path may be specified. If it does not exist
```

Comment:
I tend to name it `mkmp` for `make mountpoint` but `mkpath` is also fine. The description I'd change to:
```
Ensures that the specified mount point exists. If it does not exist in the current vfs, the directory structure is created.
```